### PR TITLE
[develop] Feature - Refactor code

### PR DESCRIPTION
**Pull request by luqp**

### CODE REVIEW

**Review by nanrehd — APPROVED** [review]

**Comment by nanrehd** [comment]
We have to keep in mind that we'll need to add some more changes (on hooks and runner classes)

File: src/main/java/org/fundacionjala/trello/pages/core/WebObject.java [hunk 1]
```diff
@@ -11,7 +12,7 @@
  */
 public abstract class WebObject {
 
-    private static final int SECONDS = 20;
+    private static final int SECONDS = Environment.getInstance().getExplicitTimeWait();
```
Comment (carlosmc23) [inline src/main/java/org/fundacionjala/trello/pages/core/WebObject.java:15]: I think that the Class DriverFactory should be in charge to get al the properties of the Environment, and give to webObjec through an Instance

**Review by carlosmc23 — COMMENTED** [review]
Good Job, I have some comments that you could review

**Comment by nanrehd** [comment]
having tried a bit more, maybe using the tags `@quitDriver` and adding that as value on the hook definition would be a better idea for executing each test separately(as a help when developing the tests) and when executing all tests together these should not be present(as your commit shows)

**Review by Snip-77 — APPROVED** [review]
Looks good for merge for me. But I think Carlos And Diego leaved comments.

File: src/main/java/org/fundacionjala/trello/pages/core/WebObject.java (hunk 1 above)
Comment (luqp) [reply src/main/java/org/fundacionjala/trello/pages/core/WebObject.java:15]: I'm sorry, I didn't understand your objective to do that. And I have some questions:
Why DriverFactory should have the explicit time property value?
Why does it need to wrap environment class work?

Comment (luqp) [reply src/main/java/org/fundacionjala/trello/pages/core/WebObject.java:15]: I'm sorry, I didn't understand your objective to do that. And I have some questions:
Why DriverFactory should have the explicit time property value?
Why does it need to wrap environment class work?

**Comment by luqp** [comment]
> having tried a bit more, maybe using the tags `@quitDriver` and adding that as value on the hook definition would be a better idea for executing each test separately(as a help when developing the tests) and when executing all tests together these should not be present(as your commit shows)

@nanrehd, That is a nice solution, but I'm not sure how to reach this without duplicate code.
Could you give an example?

**Pull request merged**